mara_pipelines/pipelines.py

```diff
@@ -81,24 +81,49 @@ def html_doc_items(self) -> List[Tuple[str, str]]:
 
 
 class Task(Node):
-    def __init__(self, id: str, description: str, commands: Optional[List[Command]] = None, max_retries: Optional[int] = None) -> None:
+    def __init__(self, id: str, description: str, commands: Optional[Union[Callable, List[Command]]] = None, max_retries: Optional[int] = None) -> None:
         super().__init__(id, description)
-        self.commands = []
+        self.callable_commands = callable(commands)
         self.max_retries = max_retries
 
-        for command in commands or []:
-            self.add_command(command)
-
-    def add_command(self, command: Command, prepend=False):
+        if self.callable_commands:
+            self._commands = None
+            self.__commands_callable = commands
```

Comment:
I mildly dislike this name because it does't really say what it does. Maybe `commands_generator_func`?  `dynamic_commands_generator_func`?

Reply:
is now renamed to `__dynamic_commands_generator_func`. I still kept it private tho. I don't know if someone should by design assign the function after construction.